### PR TITLE
Add change directory to reservations spec 2020-10-01-preview

### PR DESCRIPTION
**Pull request by corquiri**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/examples/ChangeDirectoryReservationOrder.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    "reservationOrderId": "a075419f-44cc-497f-b68a-14ee811d48b9",
+    "body": {
+      "destinationTenantId": "906655ea-30be-4587-9d12-b50e077b0f32"
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "reservationOrder" : {
+          "id": "1f14354c-dc12-4c8d-8090-6f295a3a34aa",
```
Comment (varunkumta) [inline specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/examples/ChangeDirectoryReservationOrder.json:13]: should this match the id in line 4?

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json [hunk 2]
```diff
@@ -322,6 +322,49 @@
         }
       }
     },
+    "/providers/Microsoft.Capacity/reservationOrders/{reservationOrderId}/changeDirectory": {
```
Comment (varunkumta) [inline specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json:325]: how do you validate that the current identity/caller of this api is authorized to make this change? you check the caller's permission on the destination tenant?

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json [hunk 3]
```diff
@@ -322,6 +322,49 @@
...
+          }
+        ],
+        "tags": [
+          "ReservationOrder, ChangeDirectory"
+        ],
+        "responses": {
+          "200": {
+            "description": "List of `Reservation`s created after the split operation.",
+            "schema": {
+              "type": "array",
```
Comment (varunkumta) [inline specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json:353]: as per your example this should not be an array

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json [hunk 4]
```diff
@@ -322,6 +322,49 @@
...
+          {
+            "$ref": "#/parameters/ChangeDirectoryParameter"
+          }
+        ],
+        "tags": [
+          "ReservationOrder, ChangeDirectory"
+        ],
+        "responses": {
+          "200": {
+            "description": "List of `Reservation`s created after the split operation.",
```
Comment (varunkumta) [inline specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json:351]: fix text

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json [hunk 5]
```diff
@@ -1482,6 +1525,59 @@
...
+            }
+          }
+        },
+        "reservations": {
+          "type": "array",
+          "items": {
+            "type": "object",
+            "description": "Change direcotry result for reservation",
+            "properties": {
+              "id": {
```
Comment (varunkumta) [inline specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json:1560]: if the schema of the reservations result and reservationorder result is meant to be the same, consider a separate definition for it and ref'ing it from these 2

**Review by varunkumta — COMMENTED** [review]
please take a look at review comments

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/examples/ChangeDirectoryReservationOrder.json (hunk 1 above)
Comment (corquiri) [reply specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/examples/ChangeDirectoryReservationOrder.json:13]: Will correct this to match id in this example.

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json (hunk 2 above)
Comment (corquiri) [reply specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json:325]: the validation happens in the backend. Error response is thrown if the caller does not have permission in the destination tenant

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json (hunk 4 above)
Comment (corquiri) [reply specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json:351]: will update

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json (hunk 3 above)
Comment (corquiri) [reply specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json:353]: will fix

File: specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json (hunk 5 above)
Comment (corquiri) [reply specification/reservations/resource-manager/Microsoft.Capacity/preview/2020-10-01-preview/reservations.json:1560]: will use ref

**Comment by corquiri** [comment]
CI is complaining about 
1038 - AddedPath
but that is whole purpose of this PR. I want to add a new path. Can I ignore this error ?

**Comment by ChenTanyi** [comment]
@corquiri Yes, you can ignore it. The new API only need ARM signoff. @jianyexi I think new API couldn't mark as breaking change, due to it doesn't affect any current SDK.

**Comment by jianyexi** [comment]
> @corquiri Yes, you can ignore it. The new API only need ARM signoff. @jianyexi I think new API couldn't mark as breaking change, due to it doesn't affect any current SDK.

Adding a path requires a new api version now, you can check out the guide http://aka.ms/bcforapi

**Comment by corquiri** [comment]
@jianyexi even for preview version ? 
What should I do to create new api version?

**Comment by jianyexi** [comment]
> @jianyexi even for preview version ?
> What should I do to create new api version?

Usually the service side needs to support the new API version first, then you can add a new version swagger

**Comment by corquiri** [comment]
@jianyexi @ChenTanyi  I've moved this API to new api-version (preview) and including them in the new tag package-2020-11-preview. Now this shouldn't be a breaking change, correct?

**Comment by ChenTanyi** [comment]
@corquiri Yes, you can see the CI. The CI will tell you if there is breaking change.

**Comment by corquiri** [comment]
> @corquiri Yes, you can see the CI. The CI will tell you if there is breaking change.

Thanks. What is the next step in this PR ?

**Review by pilor — APPROVED** [review]

**Comment by corquiri** [comment]
@jianyexi @ChenTanyi now that we have ARM sign-off, can you please help in merging this PR ?

File: specification/reservations/resource-manager/readme.md [hunk 6]
```diff
@@ -29,6 +29,17 @@ openapi-type: arm
 tag: package-2020-10-preview
```
Comment (ChenTanyi) [inline specification/reservations/resource-manager/readme.md:29]: Would you want to change the default tag to your new tag?

**Comment by ChenTanyi** [comment]
@corquiri Please reply or resolve my review. Then I will merge it.

File: specification/reservations/resource-manager/readme.md (hunk 6 above)
Comment (corquiri) [reply specification/reservations/resource-manager/readme.md]: I've updated the default tag now.

**Comment by corquiri** [comment]
> @corquiri Please reply or resolve my review. Then I will merge it.

@ChenTanyi I've replied and resolved the review. Please proceed with the merge.

**Pull request merged**